aws/aws-service-bundle/src/main/java/software/amazon/smithy/java/aws/servicebundle/provider/AwsServiceBundle.java

```diff
@@ -30,16 +34,37 @@ final class AwsServiceBundle implements BundlePlugin {
     }
 
     @Override
-    public RequestOverrideConfig.Builder buildOverride(Document args) {
-        var input = args.asShape(PreRequest.builder());
-        var endpoint = URI.create(Objects.requireNonNull(serviceMetadata.getEndpoints().get(input.getAwsRegion()),
-                "no endpoint for region " + input.getAwsRegion()));
-        var identityResolver = new SdkCredentialsResolver(ProfileCredentialsProvider.create(input.getAwsProfileName()));
-        return RequestOverrideConfig.builder()
-                .putConfig(RegionSetting.REGION, input.getAwsRegion())
-                .endpointResolver(EndpointResolver.staticEndpoint(endpoint))
-                .addIdentityResolver(identityResolver)
-                .authSchemeResolver(StaticAuthSchemeResolver.getInstance())
-                .putSupportedAuthSchemes(StaticAuthSchemeResolver.staticScheme(authScheme));
+    public <C extends Client, B extends Client.Builder<C, B>> B configureClient(B clientBuilder) {
+        clientBuilder.addInterceptor(new AwsServiceClientInterceptor(serviceMetadata, authScheme));
+        clientBuilder.endpointResolver(EndpointResolver.staticEndpoint("http://dummyurl.com"));
```

Comment:
use localhost. let's avoid using anything that could result in traffic being sent to a domain squatter.